### PR TITLE
10216 missing previous document refactor2

### PR DESCRIPTION
**Pull request by mmarcotte**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -220,6 +220,7 @@
     "test:client:integration:cognito-local:ci": "npm run test:client:_integration -- --coverageDirectory=./coverage-integration-${CIRCLE_NODE_INDEX} ${TESTFILES}",
     "test:client:integration:ci": "npm run test:client:_integration -- --coverageDirectory=./coverage-integration-${CIRCLE_NODE_INDEX} ${TESTFILES}",
     "test:client:unit": "jest --config web-client/jest-unit.config.ts --testMatch '**/web-client/src/**/?(*.)+(spec|test).[jt]s?(x)'",
+    "test:client:unit:file": "jest --config web-client/jest-unit.config.ts --testMatch '**/web-client/src/**/?(*.)+(spec|test).[jt]s?(x)' --coverage=false",
```
Comment (jasonfu9966) [inline package.json:223]: thank you for these.

**Review by jasonfu9966 — APPROVED** [review]
looks good to me!

File: package.json (hunk 1 above)
Comment (zachrog) [reply package.json:223]: Looks good but I do not think you need the --testMatch flag as this script looks like it is intended to be run on a single file.

File: web-client/src/presenter/computeds/formattedDocketEntries.ts [hunk 2]
```diff
@@ -277,22 +201,27 @@ export const formattedDocketEntries = (
...
+    state.featureFlags[
+      ALLOWLIST_FEATURE_FLAGS.DOCUMENT_VISIBILITY_POLICY_CHANGE_DATE.key
+    ],
+  );
+  const visibilityPolicyDateFormatted = applicationContext
+    .getUtilities()
+    .prepareDateFromString(DOCUMENT_VISIBILITY_POLICY_CHANGE_DATE)
+    .toISO();
 
+  // now do stuff
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/computeds/formattedDocketEntries.ts]: Stuff has been done!

**Review by TomElliottFlexion — APPROVED** [review]

File: shared/src/business/entities/DocketEntry.ts [hunk 3]
```diff
@@ -466,15 +480,194 @@ export class DocketEntry extends JoiValidationEntity {
...
-  static isCourtIssued(eventCode: string): boolean {
+  static TRANSCRIPT_AGE_DAYS_MIN = 90;
+
+  static isCourtIssued({ eventCode }: { eventCode: string }): boolean {
     return COURT_ISSUED_EVENT_CODES.map(
       ({ eventCode: courtIssuedEventCode }) => courtIssuedEventCode,
     ).includes(eventCode);
   }
 
+  static isAvailableToExternal = (entry: RawDocketEntry): boolean => {
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.ts:500]: To make sure I'm thinking the same thing, 'external' here means someone who is logged in to DAWSON but does not work for the court?

File: shared/src/business/entities/DocketEntry.ts [hunk 4]
```diff
@@ -466,15 +480,194 @@ export class DocketEntry extends JoiValidationEntity {
...
+    const servedOrUnservable =
+      DocketEntry.isServed(entry) || DocketEntry.isUnservable(entry);
+
+    return (
+      servedOrUnservable &&
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.ts:505]: What about if the docket entry has been sealed?

File: shared/src/business/entities/DocketEntry.ts (hunk 3 above)
Comment (mmarcotte) [reply shared/src/business/entities/DocketEntry.ts:500]: External = non Court User

File: shared/src/business/entities/DocketEntry.ts (hunk 4 above)
Comment (mmarcotte) [reply shared/src/business/entities/DocketEntry.ts:505]: Some sealed docket entries are technically available to some External users.

File: shared/src/business/entities/DocketEntry.isPublic.test.ts [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+            expect(isPublic).toEqual(false);
+          });
+        },
+      );
+
+      describe.each(['SDEC', 'AMBR'])(
+        'Amicus Briefs and Stip Decisions',
+        eventCode => {
+          it('returns true', () => {
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.isPublic.test.ts]: "should always be visible to the public"?

File: shared/src/business/entities/DocketEntry.ts [hunk 6]
```diff
@@ -466,15 +480,194 @@ export class DocketEntry extends JoiValidationEntity {
...
+      ...[
+        ...DOCUMENT_EXTERNAL_CATEGORIES_MAP['Simultaneous Brief'],
+        ...DOCUMENT_EXTERNAL_CATEGORIES_MAP['Seriatim Brief'],
+      ].map(document => document.documentType),
+    ];
+
+    return !!documentTypes.includes(documentType);
+  }
+
+  static meetsAgeRequirements = doc => {
```
Comment (zachrog) [inline shared/src/business/entities/DocketEntry.ts]: What does this meet the age requirements for?

File: shared/src/business/entities/DocketEntry.ts [hunk 7]
```diff
@@ -466,15 +480,194 @@ export class DocketEntry extends JoiValidationEntity {
...
+      userHasAccessToCase = false,
+    }: {
+      isCourtUser?: boolean;
+      userHasAccessToCase?: boolean;
+      isPublic?: boolean;
+    },
+  ): boolean => {
+    if (!entry.isFileAttached) return false;
+    if (isCourtUser) return true;
+    if (!DocketEntry.isServed(entry) && !DocketEntry.isUnservable(entry))
```
Comment (zachrog) [inline shared/src/business/entities/DocketEntry.ts]: nit: Using negations in the if can make it harder to read for me. Is it possible to write the inverse of this without negations?

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts [hunk 8]
```diff
@@ -175,11 +173,10 @@ const handleIrsSuperUser = ({
 };
 
 const handleCourtIssued = ({ docketEntryEntity, userAssociatedWithCase }) => {
-  const isUnservable = UNSERVABLE_EVENT_CODES.includes(
-    docketEntryEntity.eventCode,
-  );
-
-  if (!DocketEntry.isServed(docketEntryEntity) && !isUnservable) {
+  if (
```
Comment (zachrog) [inline shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts]: Can this be replaced with the new function you made DocketEntry.isDownloadable()?

File: shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.ts [hunk 9]
```diff
@@ -34,12 +31,17 @@ export const getPublicDownloadPolicyUrlInteractor = async (
   if (!docketEntryEntity) {
```
Comment (zachrog) [inline shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.ts:31]: Similar to getDownloadPolicy, can all of these if statements for access now be replaced with your new function DocketEntry.isDownloadable()?

File: web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts [hunk 10]
```diff
@@ -1,83 +1,15 @@
...
-  isDocumentBriefType,
 } from '../../../../../shared/src/business/entities/EntityConstants';
 import { ClientApplicationContext } from '@web-client/applicationContext';
 import { DocketEntry } from '../../../../../shared/src/business/entities/DocketEntry';
 import { Get } from 'cerebral';
 import { getFilingsAndProceedings } from '../../../../../shared/src/business/utilities/getFormattedCaseDetail';
-import { prepareDateFromString } from '../../../../../shared/src/business/utilities/DateHandler';
 import { state } from '@web-client/presenter/app-public.cerebral';
 
-export const getMeetsPolicyChangeRequirements = (
```
Comment (rachelschneiderman) [inline web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts:23]: 😄

**Review by rachelschneiderman — APPROVED** [review]

File: web-client/src/presenter/computeds/formattedDocketEntries.ts [hunk 11]
```diff
@@ -48,52 +43,6 @@ export const setupIconsToDisplay = ({ formattedResult, isExternalUser }) => {
...
-  isSealedToExternal,
-  isServed,
-  isStipDecision,
-  isStricken,
-  isUnservable,
-  meetsPolicyChangeRequirements,
-  userHasAccessToCase,
-  userHasNoAccessToDocument,
-}) => {
-  if (!hasDocument) return false;
```
Comment (zachrog) [inline web-client/src/presenter/computeds/formattedDocketEntries.ts:69]: 🥳

**Review by zachrog — CHANGES_REQUESTED** [review]
Primarily requesting changes as I am wondering if all of the if() conditions in the getDownloadPolicy can now be replaced with the isDownloadable.

If they cannot be replaced let me know and I am good with all of the changes. Let me know if you have questions!

File: shared/src/business/entities/DocketEntry.ts (hunk 6 above)
Comment (mmarcotte) [reply shared/src/business/entities/DocketEntry.ts]: yea i found this confusing, too. and i was perpetuating the existing nomenclature. i'll rename for better ease of understanding. 

this requirement is for a transcript to be on the record for a certain amount of time -- ie it meets the age requirements to be visible/downloading

File: package.json (hunk 1 above)
Comment (mmarcotte) [reply package.json:223]: Hmm, it appears to work on the file that I specify. Perhaps chat about this more online?

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts (hunk 8 above)
Comment (mmarcotte) [reply shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts]: It's not straightforward due to the various Error messages that the interactors currently return. Not that they're being made great use of by the client. A generic error would suffice. Will think on this.

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts [hunk 12]
```diff
@@ -40,167 +30,57 @@ export const getDownloadPolicyUrlInteractor = async (
...
-    throw new UnauthorizedError(UNAUTHORIZED_DOCUMENT_MESSAGE);
-  } else if (docketEntryEntity.isSealed) {
     if (
-      docketEntryEntity.sealedTo === DOCKET_ENTRY_SEALED_TO_TYPES.PUBLIC &&
-      !userAssociatedWithCase
-    ) {
-      throw new UnauthorizedError(UNAUTHORIZED_DOCUMENT_MESSAGE);
-    } else if (
-      docketEntryEntity.sealedTo === DOCKET_ENTRY_SEALED_TO_TYPES.EXTERNAL
+      !DocketEntry.isDownloadable(docketEntryEntity, {
```
Comment (zachrog) [inline shared/src/business/useCases/getDownloadPolicyUrlInteractor.ts:68]: YAY!

File: shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.ts [hunk 13]
```diff
@@ -49,32 +60,22 @@ export const getPublicDownloadPolicyUrlInteractor = async (
...
-    docketEntryEntity,
-    applicationContext
-      .getUtilities()
-      .createISODateString(documentVisibilityChangeDate, 'yyyy-MM-dd'),
-  );
-
-  if (!isTerminalUser && isPrivate) {
+  if (
+    !DocketEntry.isDownloadable(docketEntryEntity, {
+      isTerminalUser,
```
Comment (zachrog) [inline shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.ts:65]: Yay!

**Review by zachrog — APPROVED** [review]

**Pull request merged**